### PR TITLE
test: improve coverage

### PR DESCRIPTION
**Pull request by al3mart**
Improves coverage for
- `terminate_node`
- `check_contracts_node_and_prompt`

In `pop-cli/common/contracts.rs`

Refactors a few functions so that they receive the Cli as a parameter following the changes in #315  to facilitate testing.

### CODE REVIEW

File: crates/pop-cli/src/common/contracts.rs [hunk 1]
```diff
@@ -138,4 +155,40 @@ mod tests {
...
+			.expect_warning("⚠️ The substrate-contracts-node binary is not found.");
+
+		let node_path = check_contracts_node_and_prompt(&mut cli, false).await?;
+		// Binary path is at least equals cache path + "substrate-contracts-node".
+		assert!(node_path
+			.to_str()
+			.unwrap()
+			.starts_with(&cache_path.join("substrate-contracts-node").to_str().unwrap()));
+		cli.verify()?;
+		Ok(())
```
[the remark this answers is not in the document]
Comment (AlexD10S) [reply crates/pop-cli/src/common/contracts.rs:174]: Same in the test below

File: crates/pop-cli/src/common/contracts.rs [hunk 2]
```diff
@@ -138,4 +155,40 @@ mod tests {
...
+	#[tokio::test]
+	async fn check_contracts_node_and_prompt_works() -> anyhow::Result<()> {
+		let cache_path: PathBuf = crate::cache()?;
+		let mut cli = MockCli::new()
+			.expect_warning("⚠️ The substrate-contracts-node binary is not found.")
+			.expect_confirm("📦 Would you like to source it automatically now?", true)
+			.expect_warning("⚠️ The substrate-contracts-node binary is not found.");
+
+		let node_path = check_contracts_node_and_prompt(&mut cli, false).await?;
+		// Binary path is at least equals cache path + "substrate-contracts-node".
```
Comment (peterwht) [inline crates/pop-cli/src/common/contracts.rs:168]:
```suggestion
		// Binary path is at least equal to the cache path + "substrate-contracts-node".
```

**Review by peterwht — CHANGES_REQUESTED** [review]
Nice! Just one small sentence change please :)

File: crates/pop-cli/src/common/contracts.rs (hunk 2 above)
Comment (al3mart) [reply crates/pop-cli/src/common/contracts.rs]: [0c000b6](https://github.com/r0gue-io/pop-cli/pull/391/commits/0c000b64b9e47a484a02251fd24e7ae4f6496b8f)

File: crates/pop-cli/src/common/contracts.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (al3mart) [reply crates/pop-cli/src/common/contracts.rs]: [6a23175](https://github.com/r0gue-io/pop-cli/pull/391/commits/6a23175cbd45fa6b6fb92335d426f652b74d6484)

File: crates/pop-cli/src/common/contracts.rs [hunk 3]
```diff
@@ -138,4 +151,40 @@ mod tests {
 		assert!(has_contract_been_built(Some(&path.join(name))));
 		Ok(())
 	}
+
+	#[tokio::test]
+	async fn check_contracts_node_and_prompt_works() -> anyhow::Result<()> {
+		//let cache_path: PathBuf = crate::cache()?;
+
+		let cache_path = tempfile::tempdir().expect("Could create temp dir");
```
Comment (al3mart) [inline crates/pop-cli/src/common/contracts.rs:159]: I'm afraid the cache is going to be a problem in CI. `check_contracts_node_and_prompt` at line 165 returns the cache directory, so can't really just go with a temp.

Comment (al3mart) [reply crates/pop-cli/src/common/contracts.rs:157]: Applied refactor as in https://github.com/r0gue-io/pop-cli/pull/315/files#diff-5110116ab6696521a83af6b4954dacf695aeb5e3f2aa5cd7c2c2755b1a63ec64R15

**Review by peterwht — APPROVED** [review]

**Review by AlexD10S — CHANGES_REQUESTED** [review]
Just one very small point, update the docs in `check_contracts_node_and_prompt` with the new arguments

**Review by AlexD10S — APPROVED** [review]
Thanks for adding this changes!

**Pull request merged**